File: two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int size = nums.size();
        //int a,b,c=0;
        int k=0;
        vector<int> x;
        for(int i=0;i<size;i++)
        {
            for(int j=0;j<size;j++)
            {
                if(i==j)
                    continue;
                if((nums[i]+nums[j])==target)
                {
                 k+=1;
                 x.push_back(i);
                 x.push_back(j);   
                 break;   
                 }  
                
            }
            if(k>0)
                    break;
        }
        return x;
        
    }
};

File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        
        if nums is None or len(nums) == 0:
            return None

        map_d = dict()
        result=[]
        for i in range(len(nums)):
            
            compliment = target - nums[i]

            if compliment in map_d:
                result.extend([ i, map_d[compliment]])
                break

            map_d[nums[i]] = i

        return result 